AEPIdentity/Sources/IdentityState.swift

```diff
@@ -205,11 +237,10 @@ class IdentityState {
         if identityProperties.ecid != nil, !hasIds, !hasDpids, !needResync {
             Log.trace(label: "\(LOG_TAG):\(#function)", "Not syncing identifiers at this time, no new identifiers or previously synced.")
             syncForIds = false
-        } else if identityProperties.ecid == nil {
-            Log.trace(label: "\(LOG_TAG):\(#function)", "Generating new ECID, ECID not found in persistence.")
-            identityProperties.ecid = ECID()
         }
 
+        generateAndPersistECID()
```

Comment:
there might be a slight change in logic here since this is no longer on the else branch. Should we keep 
``` 
if identityProperties.ecid != nil, !hasIds, !hasDpids, !needResync {
... 
} else {
   generateAndPersistECID()
}

Reply:
I was thinking it will generate ecid if there was none before and doesnt regenerate if already present. I was trying to simplify the logic, since if ecid is nil it is going to anyways fall to this code.